packages/shopware-6-client/src/apiService.ts

```diff
@@ -61,27 +56,30 @@ export function createInstance(
   }
 
   function onConfigChange(fn: (context: ConfigChangedArgs) => void): void {
-    console.error("++ Added config method", x);
     callbackMethods.push(fn);
   }
 
-  function configChanged(): void {
-    // TODO remove after tests
-    if (!callbackMethods.length) {
-      console.error("HAVE NO CALLBACK METHODS", x);
-    }
-    callbackMethods.forEach((fn) => fn({ config: clientConfig }));
-  }
-
   const setup = function (config: ClientSettings = {}): void {
     clientConfig = Object.assign(clientConfig, defaultConfig, config);
     reloadConfiguration();
   };
   setup(initialConfig);
 
-  const update = function (config: ClientSettings = {}): void {
+  const update = function (
+    config: ClientSettings,
+    responseConfig?: AxiosResponse<any>["config"]
```

Comment:
import axios, { AxiosInstance, AxiosResponse, AxiosRequestConfig } from "axios";

It may be `responseConfig?: AxiosResponse<AxiosRequestConfig>["config"]`

Reply:
done! :)
https://github.com/DivanteLtd/shopware-pwa/pull/875/commits/458e3dba90f66bd7255bdfacf040809fba4119f8